File: Launcher/src/main/java/com/aimc/launcher/MainActivity.java
package com.aimc.launcher;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

/**
 * AI Minecraft Launcher 主界面
 *
 * 提供以下功能：
 * 1. 启动 Minecraft 游戏 (通过 FCL)
 * 2. 启动/停止 AI 控制器
 * 3. 查看游戏状态
 */
public class MainActivity extends AppCompatActivity {

    private Button btnStartMinecraft;
    private Button btnStartAi;
    private Button btnStopAi;
    private Button btnViewLogs;
    private Button btnRefreshStatus;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        // 启动 Minecraft 按钮
        btnStartMinecraft.setOnClickListener(v -> {
            Toast.makeText(this, "正在启动 Minecraft...", Toast.LENGTH_SHORT).show();
            // TODO: 调用 FCL 或其他启动器启动游戏
        });

        // 启动 AI 按钮
        btnStartAi.setOnClickListener(v -> {
            // 从 Intent 或 SharedPreferences 读取配置
            android.contentSharedPreferences prefs =
                getSharedPreferences("ai_config", MODE_PRIVATE);
            String apiKey = prefs.getString("openai_api_key", "") ?: "";
            String task = prefs.getString("task",
                "Explore the world, mine resources, and survive.") ?: "";
            String model = prefs.getString("llm_model", "gpt-4o-mini") ?: "gpt-4o-mini";
            int port = prefs.getInt("bridge_port", 25580);

            Intent intent = new Intent(this, AiControllerService.class);
            intent.setAction(AiControllerService.ACTION_START);
            intent.putExtra(AiControllerService.EXTRA_API_KEY, apiKey);
            intent.putExtra(AiControllerService.EXTRA_TASK, task);
            intent.putExtra(AiControllerService.EXTRA_MODEL, model);
            intent.putExtra(AiControllerService.EXTRA_PORT, port);
            startService(intent);

            btnStartAi.setEnabled(false);
            btnStopAi.setEnabled(true);
            Toast.makeText(this, "AI 控制器已启动", Toast.LENGTH_SHORT).show();
        });

        // 停止 AI 按钮
        btnStopAi.setOnClickListener(v -> {
            Intent intent = new Intent(this, AiControllerService.class);
            intent.setAction(AiControllerService.ACTION_STOP);
            stopService(intent);

            btnStartAi.setEnabled(true);
            btnStopAi.setEnabled(false);
            Toast.makeText(this, "AI 控制器已停止", Toast.LENGTH_SHORT).show();
        });

        // 查看日志按钮
        btnViewLogs.setOnClickListener(v -> {
            Toast.makeText(this, "日志功能待实现", Toast.LENGTH_SHORT).show();
        });

        // 刷新状态按钮
        btnRefreshStatus.setOnClickListener(v -> {
            Toast.makeText(this, "状态已刷新", Toast.LENGTH_SHORT).show();
        });
    }

    private void initViews() {
        btnStartMinecraft = findViewById(R.id.btn_start_minecraft);
        btnStartAi = findViewById(R.id.btn_start_ai);
        btnStopAi = findViewById(R.id.btn_stop_ai);
        btnViewLogs = findViewById(R.id.btn_view_logs);
        btnRefreshStatus = findViewById(R.id.btn_refresh_status);
    }
}

File: Launcher/src/main/java/com/aimc/launcher/MainActivity.kt
package com.aimc.launcher

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.aimc.launcher.service.AiControllerService
import com.tungsten.fcl.FCLApplication
import com.tungsten.fcl.FCLRepository

/**
 * AI Minecraft Launcher 主 Activity (增强版)
 *
 * 作为应用入口，提供进入 FCL 启动器和 AI 控制器的入口。
 */
class MainActivity : AppCompatActivity() {

    private lateinit var btnStartMinecraft: Button
    private lateinit var btnStartAi: Button
    private lateinit var btnStopAi: Button
    private lateinit var btnSettings: Button
    private lateinit var btnGameDirectory: Button
    private lateinit var btnModSearch: Button
    private lateinit var btnAccount: Button
    private lateinit var tvAiStatus: TextView
    private lateinit var tvGameDir: TextView
    private var isAiRunning = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
        setupListeners()
        updateGameDirInfo()
    }

    override fun onResume() {
        super.onResume()
        updateGameDirInfo()
    }

    private fun initViews() {
        btnStartMinecraft = findViewById(R.id.btn_start_minecraft)
        btnStartAi = findViewById(R.id.btn_start_ai)
        btnStopAi = findViewById(R.id.btn_stop_ai)
        btnSettings = findViewById(R.id.btn_settings)
        btnGameDirectory = findViewById(R.id.btn_game_directory)
        btnModSearch = findViewById(R.id.btn_mod_search)
        btnAccount = findViewById(R.id.btn_account)
        tvAiStatus = findViewById(R.id.tv_ai_status)
        tvGameDir = findViewById(R.id.tv_game_dir)
    }

    private fun setupListeners() {
        // 进入 FCL 启动器主界面
        btnStartMinecraft.setOnClickListener {
            try {
                val intent = Intent(this, Class.forName("com.tungsten.fcl.activity.MainActivity"))
                startActivity(intent)
            } catch (e: Exception) {
                Toast.makeText(this, "无法启动: ${e.message}", Toast.LENGTH_LONG).show()
            }
        }

        // AI 控制器
        btnStartAi.setOnClickListener { startAiController() }
        btnStopAi.setOnClickListener { stopAiController() }

        // 设置
        btnSettings.setOnClickListener {
            startActivity(Intent(this, SettingsActivity::class.java))
        }

        // 游戏目录
        btnGameDirectory.setOnClickListener {
            try {
                val intent = Intent(this, Class.forName("com.tungsten.fcl.activity.GameDirectoryActivity"))
                startActivity(intent)
            } catch (e: Exception) {
                Toast.makeText(this, "无法打开: ${e.message}", Toast.LENGTH_SHORT).show()
            }
        }

        // Mod 搜索
        btnModSearch.setOnClickListener {
            try {
                val intent = Intent(this, Class.forName("com.tungsten.fcl.activity.ModSearchActivity"))
                startActivity(intent)
            } catch (e: Exception) {
                Toast.makeText(this, "无法打开: ${e.message}", Toast.LENGTH_SHORT).show()
            }
        }

        // 账户管理
        btnAccount.setOnClickListener {
            try {
                val intent = Intent(this, Class.forName("com.tungsten.fcl.activity.AccountActivity"))
                startActivity(intent)
            } catch (e: Exception) {
                Toast.makeText(this, "无法打开: ${e.message}", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun updateGameDirInfo() {
        try {
            val repo: FCLRepository = FCLApplication.getInstance().repository
            val versionsDir = repo.versionsDir
            val versionCount = if (versionsDir.exists() && versionsDir.isDirectory) {
                versionsDir.listFiles { f -> f.isDirectory }?.size ?: 0
            } else {
                0
            }
            tvGameDir.text = "游戏目录: ${repo.gameDirectoryPath}\n已安装版本: $versionCount"
        } catch (e: Exception) {
            tvGameDir.text = "游戏目录: 未初始化"
        }
    }

    private fun startAiController() {
        val intent = Intent(this, AiControllerService::class.java)
        startService(intent)
        Toast.makeText(this, "AI 控制器已启动", Toast.LENGTH_SHORT).show()
        btnStartAi.isEnabled = false
        btnStopAi.isEnabled = true
        isAiRunning = true
        tvAiStatus.text = getString(R.string.ai_status_running)
    }

    private fun stopAiController() {
        val intent = Intent(this, AiControllerService::class.java)
        stopService(intent)
        Toast.makeText(this, "AI 控制器已停止", Toast.LENGTH_SHORT).show()
        btnStartAi.isEnabled = true
        btnStopAi.isEnabled = false
        isAiRunning = false
        tvAiStatus.text = getString(R.string.ai_status_stopped)
    }

    override fun onDestroy() {
        super.onDestroy()
        if (isAiRunning) stopAiController()
    }
}
